index.js

```diff
@@ -28,101 +24,90 @@ function parseLocale(x) {
 }
 
 function getLocale(str) {
-	return (str && str.replace(/[.:].*/, '')) || fallback();
+	return (str && str.replace(/["]?(.*)[.:].*/, '$1')) || fallback();
 }
 
-module.exports = function (opts, cb) {
-	if (typeof opts === 'function') {
-		cb = opts;
-		opts = defaultOpts;
-	} else {
-		opts = opts || defaultOpts;
-	}
+module.exports = mem(opts => {
+	opts = opts || defaultOpts;
 
-	if (cache || getEnvLocale() || opts.spawn === false) {
-		setImmediate(cb, null, cache);
-		return;
-	}
+	const envLocale = getEnvLocale();
+	let locale;
```

Comment:
This looks moot. It's not being used anywhere. It's more like the old design which is no longer necessary. Line 37 can also be inlined into line 38.